spec/features/applications/show_spec.rb

```diff
@@ -0,0 +1,197 @@
+
+        name: "Joe Smith", 
+        street_address: "2323 Wysteria Ln.",
+        city: "Littleton",
+        state: "CO",
+        zip_code: "89321",
+        description: "Love pets; live on farm",
+        application_status: "In Progress"
+      })
+
+      visit "/applications/#{application.id}"
+
+      expect(page).to have_content(application.name)
+      expect(page).to have_content(application.street_address)
+      expect(page).to have_content(application.city)
+      expect(page).to have_content(application.state)
+      expect(page).to have_content(application.zip_code)
+      expect(page).to have_content(application.description)
+      expect(page).to have_content(application.application_status)
+    end
+
+    it "has a section to search for pets by name" do
+      shelter = Shelter.create(name: "Aurora shelter", city: "Aurora, CO", foster_program: false, rank: 9)
+
+      pet_1 = Pet.create(adoptable: true, age: 1, breed: "sphynx", name: "Lucille Bald", shelter_id: shelter.id)
+      pet_2 = Pet.create(adoptable: true, age: 3, breed: "doberman", name: "Lobster", shelter_id: shelter.id)
+
+      application = Application.create!({
+        name: "Joe Smith", 
+        street_address: "2323 Wysteria Ln.",
+        city: "Littleton",
+        state: "CO",
+        zip_code: "89321",
+        description: "Love pets; live on farm",
+        application_status: "In Progress"
+      })
+
+      visit "/applications/#{application.id}"
+
+      expect(page).to have_content("Add a pet to this application")
+      expect(page).to have_field(:pet_name)
+      fill_in(:pet_name, with: "Lobster")
+      click_button("Search for Pets")
+
+
+      expect(current_path).to eq("/applications/#{application.id}")
+      expect(page).to have_content("Lobster")
+      expect(page).to have_button("Adopt this Pet")
+    end
+
+    it "can add a pet to an application" do
+      application = Application.create!({
+        name: "Joe Smith", 
+        street_address: "2323 Wysteria Ln.",
+        city: "Littleton",
+        state: "CO",
+        zip_code: "89321",
+        description: "Love pets; live on farm",
+        application_status: "In Progress"
+      })
+
+      shelter = Shelter.create(name: "Aurora shelter", city: "Aurora, CO", foster_program: false, rank: 9)
+
+      pet_1 = Pet.create(adoptable: true, age: 1, breed: "sphynx", name: "Lucille Bald", shelter_id: shelter.id)
+      pet_2 = Pet.create(adoptable: true, age: 3, breed: "doberman", name: "Lobster", shelter_id: shelter.id)
+
+      visit "/applications/#{application.id}"
+
+      fill_in(:pet_name, with: "Lucille Bald")
+      click_button("Search for Pets")
+      expect("Lucille Bald").to_not appear_before("Search for Pets")
+      expect(page).to have_button("Adopt this Pet")
+      click_button("Adopt this Pet")
+      expect(current_path).to eq("/applications/#{application.id}")
+      expect("Lucille Bald").to appear_before("Search for Pets")
```

Comment:
Another option here in order to make sure that Lucille Bald is in the right section of page is to use **within blocks**. Add a div or a section to your view with an ID called `associated-pets` or something, and then you can do:
```
within "#associated-pets" do
    expect(page).to have_content("Lucille Bald")
end
```